### DISCUSSION TITLE
Opening new tabs with session files possible?

### DISCUSSION THREAD

**Discussion by cd-a** [kovidgoyal/kitty#4348] [Q&A] [answered]
I know that within a session file, I can open multiple tabs.

But I want to start a new session as a tab to the currently existing ones.

Let's say I have one window open, and then start this session:

```
new_tab Shell

cd ~/dev

# Set the layout for the current tab
enabled_layouts tall:bias=76;full_size=1;mirrored=true,stack

launch zsh -i
launch zsh -i
launch zsh -i
```

Now this opens a new window instead of a new tab.

Am I right in thinking that a session always opens a new tab, and only supports tabs within itself?

If that's the case, how can I accomplish that? I found https://github.com/kovidgoyal/kitty/issues/3601 where he talks about launch and combine, is that the way to go? I haven't seen an example of that, does anyone use it like this, or am I just missing something?

**Comment by page-down** [comment, marked as answer]
> ...  start a new session as a tab ...

```conf
map f1 combine : launch --type tab : remote_control set-enabled-layouts tall,stack : launch : launch
```

Or use remote control.
Or write your custom kitten to create a new tab however you want.

To use `set-enabled-layouts`, you need the nightly version, or compile from src.

I've also thought about making `session` available for the new OS window or new Tab (e.g. with different shells presets), but I don't have any motivation.

**Reply by cd-a** [reply]
Thanks for the explanation @page-down, appreciate it. I've compiled kitty from latest master for testing, and I can now spawn tabs properly.

Still gotta figure out how to do it properly, but it's a good start.
